pkg/devfile/adapters/common/command_supervisor.go

```diff
@@ -0,0 +1,60 @@
+package common
+
+import (
+	"fmt"
+	"github.com/openshift/odo/pkg/devfile/parser/data/common"
+	"github.com/openshift/odo/pkg/machineoutput"
+)
+
+// supervisorCommand encapsulates a supervisor-specific command
+type supervisorCommand struct {
+	adapter commandExecutor
+	cmd     []string
+	info    ComponentInfo
+}
+
+// newSupervisorInitCommand creates a command that initializes the supervisor for the specified devfile if needed
+// nil is returned if no devfile-specified container needing supervisor initialization is found
+func newSupervisorInitCommand(command common.DevfileCommand, adapter commandExecutor) (command, error) {
+	cmd := []string{SupervisordBinaryPath, "-c", SupervisordConfFile, "-d"}
+	info, err := adapter.SupervisorComponentInfo(command)
+	if err != nil {
+		adapter.Logger().ReportError(err, machineoutput.TimestampNow())
+		return nil, err
+	}
+	if !info.IsEmpty() {
+		return supervisorCommand{
+			adapter: adapter,
+			cmd:     cmd,
+			info:    info,
+		}, nil
+	}
+	return nil, nil
+}
+
+// newSupervisorStopCommand creates a command implementation that stops the specified command via the supervisor
+func newSupervisorStopCommand(command common.DevfileCommand, executor commandExecutor) (command, error) {
+	cmd := []string{SupervisordBinaryPath, SupervisordCtlSubCommand, "stop", "all"}
+	if stop, err := newOverridenSimpleCommand(command, executor, cmd); err == nil {
+		// override spinner message
+		stop.msg = fmt.Sprintf("Stopping %s command %q, if running", command.GetID(), command.Exec.CommandLine)
+		return stop, err
+	} else {
+		return nil, err
+	}
+}
+
+// newSupervisorStartCommand creates a command implementation that starts the specified command via the supervisor
+func newSupervisorStartCommand(command common.DevfileCommand, cmd string, adapter commandExecutor) (command, error) {
+	cmdLine := []string{SupervisordBinaryPath, SupervisordCtlSubCommand, "start", cmd}
+	return newOverridenSimpleCommand(command, adapter, cmdLine)
+}
+
+func (s supervisorCommand) Execute(show bool) error {
+	err := ExecuteCommand(s.adapter, s.info, s.cmd, true, nil, nil)
```

Comment:
is the show value intentionally always true here?

Reply:
I'll double check but it seems that supervisor commands were always executed with show=true in the previous implementation.

Reply:
See https://github.com/openshift/odo/blob/3ca50987d0433fe4d3d74ebc63551f653d264161/pkg/devfile/adapters/docker/component/utils.go#L428 and https://github.com/openshift/odo/blob/c89c1ab1d730fb8495f91ce6ea0ed7ff8f784711/pkg/devfile/adapters/kubernetes/component/adapter.go#L547

Reply:
I tried changing it to false and running the commands, found no difference.

Reply:
OK